aocd/models.pyi

```diff
@@ -0,0 +1,79 @@
+import datetime
+
+import bs4
+import typing as t
+
+_Answer = t.Optional[t.Union[int, str]]
+_Result = t.TypedDict("_Result", {"time": datetime.timedelta, "rank": int, "score": int})
+
+class User:
+    token: t.Text
+
+    def __init__(self, token: t.Text) -> None: ...
+    @classmethod
+    def from_id(cls, id: t.Text) -> "User": ...
+    @property
+    def auth(self) -> t.Dict[t.Text, t.Text]: ...
+    @property
+    def id(self) -> t.Text: ...
+    @property
+    def memo_dir(self) -> t.Text: ...
+    def get_stats(
+        self, years: t.Optional[t.Union[t.Iterable[int], int]] = ...
+    ) -> t.Dict[t.Tuple[int, int], t.Dict[t.Literal["a", "b"], _Result]]: ...
+
+def default_user() -> User: ...
+
+class Puzzle:
+    year: int
+    day: int
+    input_data_url: t.Text
+    submit_url: t.Text
+    input_data_fname: t.Text
+    example_input_data_fname: t.Text
+    answer_a_fname: t.Text
+    answer_b_fname: t.Text
+    incorrect_answers_a_fname: t.Text
+    incorrect_answers_b_fname: t.Text
+    title_fname: t.Text
+
+    def __init__(self, year: int, day: int, user: t.Optional[User] = ...): ...
+    @property
+    def user(self) -> User: ...
+    @property
+    def input_data(self) -> t.Text: ...
+    @property
+    def example_data(self) -> t.Text: ...
+    @property
+    def title(self) -> t.Text: ...
+    @property
+    def answer_a(self) -> _Answer: ...
+    @answer_a.setter
+    def answer_a(self, val: _Answer) -> None: ...
+    @property
+    def answered_a(self) -> bool: ...
+    @property
+    def answer_b(self) -> _Answer: ...
+    @answer_b.setter
+    def answer_b(self, val: _Answer) -> None: ...
+    @property
+    def answered_b(self) -> bool: ...
+    def answered(self, part: t.Literal["a", "b"]) -> bool: ...
+    @property
+    def answers(self) -> t.Tuple[_Answer, _Answer]: ...
+    @answers.setter
+    def answers(self, val: t.Tuple[_Answer, _Answer]) -> None: ...
+    @property
+    def incorrect_answers_a(self) -> t.Dict[t.Text, t.Text]: ...
+    @property
+    def incorrect_answers_b(self) -> t.Dict[t.Text, t.Text]: ...
+    def solve(self) -> t.Tuple[_Answer, _Answer]: ...
+    @property
+    def url(self) -> t.Text: ...
+    def view(self) -> None: ...
+    # TODO: my_stats seems broken
+    @property
+    def my_stats(self) -> t.Dict[t.Literal["a", "b"], _Result]: ...
+    # TODO: should this really expose bs4 to library users?
+    @property
+    def easter_eggs(self) -> t.List[bs4.element.Tag]: ...
```

Comment:
@wimglenn these functions/properties seem... weird. `my_stats` raised exceptions when I tried to use it and `easter_eggs` exposes `bs4.element.Tag` (which imo, seems like it ought to be an implementation detail) to users.

I should probably open issues for these, but I noticed this stuff while working on the stubs, so I figured I'd just let ya know here.

Reply:
Just checked `my_stats` and it still seems to be working fine for me.  Are you sure your auth token isn't expired? Sometimes it's not obvious to tell when a token died, if aocd has already cached everything needed. `aocd-token --check` will show you which token(s) are alive and which are dead.

I've added https://github.com/wimglenn/advent-of-code-data/pull/80 to give a better error message in case the problem was a dead token.

Reply:
ah, that's probably it. I'm on a different machine right now, but I'll check when I'm back at the machine I was using to work on this.

Reply:
#80 looks mostly good, but I noticed a small mistake. There could be other places where you've done it incorrectly too. https://github.com/wimglenn/advent-of-code-data/pull/80/files#r1008985078

Reply:
That is not a mistake. If it was a mistake, the test wouldn't pass.

Reply:
I'll keep the conversation about #80 on that PR. Leaving this unresolved so I remember to check `my_stats`.